### ISSUE TITLE
Cache feedstocks repo

### ISSUE THREAD

**Issue by jakirkham** [conda-forge/conda-forge-webservices#138]
For updating the feedstocks repo, it would be nice if we cached a cloned version of the feedstocks repo as part of the build. Admittedly we would still have to run `git pull` on it each time for good measure. However it would at least have much (if not all) of the changes already pulled in. So running `git pull` shouldn't have to do very much. This would be much better than starting from scratch and cloning the feedstocks repo each time.

**Comment by isuruf** [comment]
Do a shallow clone at first?

**Comment by jakirkham** [comment]
Good idea. We probably can get away with a shallow clone. Honestly even just one commit might be fine. Would have to play with it locally to test though.

FWIW we are not cloning all of the submodules or anything (that is very much intentional). Though if we start caching the feedstocks repo, we are definitely going to have old cloned submodules hanging around (from previous runs).

That said, we are far from performance issues at this point. Looks like it takes ~15s to go from a PR being merged on a feedstock to updating the feedstocks repo. So we are doing ok. Just something to keep on the radar IMHO.

**Comment by isuruf** [comment]
Are there parallel workers in heroku configured for the repo? When a PR is merged feedstock-update and team-update are both done. So, one might be waiting for the other

**Comment by jakirkham** [comment]
IIUC that's what PR ( https://github.com/conda-forge/conda-forge-webservices/pull/76 ) was suppose to do. So it should be able to support 2-3 workers according to this [doc]( https://devcenter.heroku.com/articles/optimizing-dyno-usage#python ).

Though we are not using `gunicorn` AFAIK. We probably should start doing that. Also not sure if there were other changes we needed to make.
